wpiutil/src/main/java/edu/wpi/first/util/CircularBuffer.java

```diff
@@ -167,6 +167,15 @@ public double get(int index) {
     return m_data[(m_front + index) % m_data.length];
   }
 
+  /**
+   * Returns a copy of the currently-stored data in the buffer.
+   *
+   * @return A copy of the underlying double array.
+   */
+  public double[] getData() {
+    return m_data.clone();
+  }
```

Comment:
This should be removed. Since this is a circular buffer, the beginning of the input and output buffers in LinearFilter aren't the same as the beginning of the underlying storage. Use a for loop up to the size and get() instead.